test/test_torch.py

```diff
@@ -4342,6 +4342,85 @@ def test_diff(self, device, dtype):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
 
+    @onlyOnCPUAndCUDA
+    @dtypesIfCUDA(*set(torch.testing.get_all_math_dtypes('cuda')))
+    @dtypes(*set(torch.testing.get_all_math_dtypes('cpu')))
+    def test_gradient(self, device, dtype):
+        inputs = (
+            ((5,), [2.73], 0, 1),
```

Comment:
These cases will be OK to test scalars and list of scalars, but testing a list of tensors may require new test cases (maybe in a separate test?)